examples/connection_snippets.rs

```diff
@@ -0,0 +1,38 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+extern crate mongodb;
+
+use mongodb::error::Result;
+
+const CONNECTION_URI: &str =
+    "mongodb+srv://<username>:<password>@<cluster-url>/<dbname>?w=majority";
+
+#[cfg(feature = "sync")]
+fn main() -> Result<()> {
+    use mongodb::sync::Client;
+    let client = Client::with_uri_str(CONNECTION_URI)?;
+    let database = client.database("test");
+    Ok(())
+}
+
+#[cfg(feature = "tokio-runtime")]
+#[tokio::main]
```

Comment:
typically in docs examples, we surround the sections we want the other teams to copy/paste with comments so they know exactly what should be taken. I think we should do that here too. In order to actually connect, users need the `#[tokio::main]` thing, so we should include the entirety of the `main` declaration, but not the `cfg` part. If we put comments between the two macros does that mess anything up?

e.g.
```rust
#[cfg(feature = "tokio-runtime")]
// CONNECTION EXAMPLE STARTS HERE
#[tokio::main]
async fn main() -> Result<()> { ... }
// CONNECTION EXAMPLE ENDS HERE
```

Reply:
The other code snippets in Atlas only include the lines to create a client/database, so I added the comments in line with that; let me know if you think we should do it differently

Reply:
Oh sorry, I didn't see your response here before commenting in the other thread. I think we need to deviate a little from what other languages do here, since in Rust the `.await`s won't work unless called in the context of a `#[tokio::main]` annotated function, and the goal here is to provide copy/pastable code for new users.